src/libData/AccountData/AccountStoreSCEvm.cpp

```diff
@@ -0,0 +1,661 @@
+/*
+ * Copyright (C) 2019 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <vector>
+#include "AccountStoreSC.h"
+#include "EvmClient.h"
+#include "libUtils/EvmCallParameters.h"
+#include "libUtils/EvmJsonResponse.h"
+#include "libUtils/EvmUtils.h"
+
+template <class MAP>
+void AccountStoreSC<MAP>::EvmCallRunner(
+    INVOKE_TYPE invoke_type, EvmCallParameters& params, const uint32_t& version,
+    bool& ret, TransactionReceipt& receipt,
+    evmproj::CallResponse& evmReturnValues) {
+  auto call_already_finished{false};
+
+  auto worker = [this, &params, &invoke_type, &ret, &version,
+                 &call_already_finished, &evmReturnValues]() mutable -> void {
+    if (invoke_type == RUNNER_CREATE || invoke_type == RUNNER_CALL) {
```

Comment:
I think we don't need that invoke_type. Can probably remove the function argument.